### PR TITLE
GEODE-7930: Fix endpoint name truncation bug

### PR DESCRIPTION
**Pull request by jvarenina**
- Removed truncation of endpoint when exceedes 99 chars
  - Also, fixed wrong logging of error when endpoint is successfully
    added to the pool

Related to solution, I haven't found any limitation checks e.g. if host name exceeds full domain maximum length (see JIRA for more info) in cases that work on native client (e.g. subscription is "disabled"), so I didn't put any restrictions of that kind in addEP function.

Related to integration test, I think that it would be hard to create generic tests to verify maximum supported length of host name name (see JIRA for more info), since host name depends on environment where the tests are executed. I only tested this manually.

### CODE REVIEW

File: cppcache/src/ThinClientPoolDM.cpp [hunk 1]
```diff
@@ -1966,11 +1966,8 @@ GfErrType ThinClientPoolDM::sendRequestToEP(const TcrMessage& request,
 }
 
 TcrEndpoint* ThinClientPoolDM::addEP(ServerLocation& serverLoc) {
-  std::string serverName = serverLoc.getServerName();
-  int port = serverLoc.getPort();
-  char endpointName[100];
-  std::snprintf(endpointName, 100, "%s:%d", serverName.c_str(), port);
-
+  std::string endpointName = serverLoc.getServerName() + ":"
```
Comment (pivotal-jbarrett) [inline cppcache/src/ThinClientPoolDM.cpp:1969]: Please use `const auto`, `const auto&` or `auto` whenever possible.

**Comment by pivotal-jbarrett** [comment]
The Travis failure is formatting. Install clang-format and see the BUILDING.md on how to configure the build to use it.

File: cppcache/src/ThinClientPoolDM.cpp (hunk 1 above)
Comment (jvarenina) [reply cppcache/src/ThinClientPoolDM.cpp]: Thanks, I have fixed it.

**Comment by mkevo** [comment]
As this PR is approved, I will merge it.

**Comment by pivotal-jbarrett** [comment]
@mkevo please remember to squash and merge in the future.

**Comment by mkevo** [comment]
> @mkevo please remember to squash and merge in the future.

Thanks @pivotal-jbarrett , this is my first merge. :)

**Pull request merged**